test/spec/geo/gmaps/gmaps-cartodb-layer-group.spec.js

```diff
@@ -1,40 +1,297 @@
+            'http://2.ashbu.cartocdn.com/documentation/api/v1/map/0123456789/0/{z}/{x}/{y}.grid.json',
+            'http://3.ashbu.cartocdn.com/documentation/api/v1/map/0123456789/0/{z}/{x}/{y}.grid.json'
+          ],
+          [
+            'http://0.ashbu.cartocdn.com/documentation/api/v1/map/0123456789/1/{z}/{x}/{y}.grid.json',
+            'http://1.ashbu.cartocdn.com/documentation/api/v1/map/0123456789/1/{z}/{x}/{y}.grid.json',
+            'http://2.ashbu.cartocdn.com/documentation/api/v1/map/0123456789/1/{z}/{x}/{y}.grid.json',
+            'http://3.ashbu.cartocdn.com/documentation/api/v1/map/0123456789/1/{z}/{x}/{y}.grid.json'
+          ]
+        ],
+        'attributes': [
+          'http://ashbu.cartocdn.com/documentation/api/v1/map/0123456789/0/attributes',
+          'http://ashbu.cartocdn.com/documentation/api/v1/map/0123456789/1/attributes'
+        ]
+      },
+      indexOfLayersInWindshaft: [1, 2]
+    }, {
+      layersCollection: layersCollection
+    });
+  });
+
+  afterEach(function () {
+    document.body.removeChild(container);
   });
 
-  var layerGroupView = new GmapsCartoDBLayerGroupView(layerGroupModel, {
-    nativeMap: gmapsMap
+  it('should set the right tile url into the innter leaflet Layer', function () {
+    layerGroupView = new LeafletCartoDBLayerGroupView(layerGroupModelMock, {
+      nativeMap: googleMap,
+      mapModel: mapModelMock
+    });
+    expect(layerGroupView.options.tiles[0]).toEqual('http://0.ashbu.cartocdn.com/documentation/api/v1/map/0123456789/1,2/{z}/{x}/{y}.png');
   });
-  gmapsMap.overlayMapTypes.setAt(1, layerGroupView.gmapsLayer);
-  return layerGroupView;
-};
 
-var expectTileURLTemplateToMatch = function (layerGroupView, expectedTileURLTemplate) {
-  expect(removeSubdomainFromTileURL(expectedTileURLTemplate)).toEqual(layerGroupView.options.tiles[0]);
-};
+  describe('interactivity', function () {
+    it('should not enable interaction when there are no interactive layers ', function () {
+      layerGroupView = new LeafletCartoDBLayerGroupView(layerGroupModelMock, {
+        nativeMap: googleMap,
+        mapModel: mapModelMock
+      });
 
-var removeSubdomainFromTileURL = function (tileURL) {
-  return tileURL.replace('{s}', '0');
-};
+      expect(layerGroupView.interaction.length).toEqual(0);
+    });
+
+    it('should enable interaction when there are no interactive layers but map model has enabled interaction ', function () {
+      mapModelMock.isFeatureInteractivityEnabled.and.returnValue(true);
+
+      layerGroupView = new LeafletCartoDBLayerGroupView(layerGroupModelMock, {
+        nativeMap: googleMap,
+        mapModel: mapModelMock
+      });
+
+      expect(layerGroupView.interaction.length).toEqual(2);
+    });
+
+    it('should enable interaction when there are interactive layers ', function () {
+      mapModelMock.isFeatureInteractivityEnabled.and.returnValue(false);
+      spyOn(cartoDbLayer0, 'isInteractive').and.returnValue(true);
```

Comment:
Perhaps we can move this spy to the `beforeEach` so that we don't have to repeat it in other `it` blocks.